src/libcore/ops/drop.rs

```diff
@@ -1,85 +1,136 @@
-/// Used to run some code when a value goes out of scope.
-/// This is sometimes called a 'destructor'.
+/// Custom code within the destructor.
 ///
-/// When a value goes out of scope, it will have its `drop` method called if
-/// its type implements `Drop`. Then, any fields the value contains will also
-/// be dropped recursively.
+/// When a value is no longer needed, Rust will run a "destructor" on that value.
+/// The most common way that a value is no longer needed is when it goes out of
+/// scope. Destructors may still run in other circumstances, but we're going to
+/// focus on scope for the examples here. To learn about some of those other cases,
+/// please see [the reference] section on destructors.
 ///
-/// Because of this recursive dropping, you do not need to implement this trait
-/// unless your type needs its own destructor logic.
+/// [the reference]: https://doc.rust-lang.org/reference/destructors.html
 ///
-/// Refer to [the chapter on `Drop` in *The Rust Programming Language*][book]
-/// for some more elaboration.
+/// The `Drop` trait provides a way to customize the destructor of a type.
+/// While this is not necessary for most types, there are some cases where
+/// it is useful. Most notably when dealing with resources. That resource may
+/// be memory, it may be a file descriptor, it may be a network socket.
+/// But when the type is no longer going to be used, it should "clean up" that
+/// resource by freeing the memory or closing the file or socket. This is
+/// the job of a destructor, and therefore the job of `Drop::drop`.
 ///
-/// [book]: ../../book/ch15-03-drop.html
+/// ## Using the `Drop` trait
 ///
-/// # Examples
+/// When a value goes out of scope, the compiler automatically inserts "drop glue"
+/// for this value. This means calling to `Drop::drop` for it, and then recursively for all
```

Comment:
```suggestion
/// for this value. This means calling `Drop::drop` for it, and then recursively for all
```

Reply:
Actually I'd formulate this a bit differently:

> This means it will call `Drop::drop` on that value, and then recursively call the "drop glue" for all fields.

Reply:
(I posted another comment above which got hidden by GH when you accepted the suggestions... but anyway this is superseded by https://github.com/rust-lang/rust/pull/71598#discussion_r417826595)